target_chains/ethereum/contracts/contracts/entropy/Entropy.sol

```diff
@@ -291,7 +301,96 @@ abstract contract Entropy is IEntropy, EntropyState {
         );
 
         clearRequest(provider, sequenceNumber);
+        updateProviderCurrentCommitment(
+            provider,
+            sequenceNumber,
+            providerRevelation
+        );
+    }
+
+    // Fulfill a request for a random number and call back the requester. This method validates the provided userRandomness
+    // and provider's revelation against the corresponding commitment in the in-flight request. If both values are validated,
+    // this function calls the requester's entropyCallback method with the sequence number and the random number as arguments.
+    //
+    // Note that this function can only be called once per in-flight request. Calling this function deletes the stored
+    // request information (so that the contract doesn't use a linear amount of storage in the number of requests).
+    // If you need to use the returned random number more than once, you are responsible for storing it.
+    //
+    // Anyone can call this method to fulfill a request, but the callback will only be made to the original requester.
+    function revealAndCall(
```

Comment:
Can this function be a simple wrapper around `reveal` instead of reimplementing the same functionality piecemeal?

Reply:
There is a check on the requestor that is present in one and not in another.